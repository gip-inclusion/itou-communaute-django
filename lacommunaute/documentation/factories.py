import factory
from faker import Faker

from lacommunaute.documentation.models import Category, Document, DocumentRating


faker = Faker()


class AbstractDocumentationFactory(factory.django.DjangoModelFactory):
    name = factory.Faker("name")
    description = factory.Faker("sentence", nb_words=100)
    short_description = factory.Faker("sentence", nb_words=10)
    image = factory.django.ImageField(filename="banner.jpg")


class CategoryFactory(AbstractDocumentationFactory):
    class Meta:
        model = Category

    class Params:
        for_snapshot = factory.Trait(
            name="Test Category", description="Test description", short_description="Test short description"
        )


class DocumentFactory(AbstractDocumentationFactory):
    category = factory.SubFactory(CategoryFactory)

    class Meta:
        model = Document

    class Params:
        for_snapshot = factory.Trait(
            name="Test Document",
            description="Test description",
            short_description="Test short description",
            category=factory.SubFactory(CategoryFactory, for_snapshot=True),
        )

    @factory.post_generation
    def with_tags(self, create, extracted, **kwargs):
        if not create or not extracted:
            return

        if isinstance(extracted, list):
            for tag in extracted:
                self.tags.add(tag)


class DocumentRatingFactory(factory.django.DjangoModelFactory):
    document = factory.SubFactory(DocumentFactory)
    rating = factory.Faker("random_int", min=1, max=5)

    class Meta:
        model = DocumentRating
        skip_postgeneration_save = True

    @factory.post_generation
    def set_created(self, create, extracted, **kwargs):
        if extracted:
            self.created = extracted
            self.save()
