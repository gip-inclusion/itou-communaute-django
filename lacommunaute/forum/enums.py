from django.db import models


class Kind(models.TextChoices):
    PUBLIC_FORUM = "PUBLIC_FORUM", "Espace public"
    NEWS = "NEWS", "Actualit√©s"
