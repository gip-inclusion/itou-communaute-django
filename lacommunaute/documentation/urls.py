from django.urls import path

from lacommunaute.documentation.views import (
    CategoryCreateView,
    CategoryDetailView,
    CategoryListView,
    CategoryUpdateView,
    DocumentDetailView,
)


app_name = "documentation"


urlpatterns = [
    path("", CategoryListView.as_view(), name="category_list"),
    path("<str:slug>-<int:pk>/", CategoryDetailView.as_view(), name="category_detail"),
    path("category/create/", CategoryCreateView.as_view(), name="category_create"),
    path("<str:slug>-<int:pk>/update/", CategoryUpdateView.as_view(), name="category_update"),
    path("<int:category_pk>/<str:slug>-<int:pk>/", DocumentDetailView.as_view(), name="document_detail"),
]
