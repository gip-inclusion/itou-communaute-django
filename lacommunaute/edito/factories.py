import factory
from faker import Faker

from lacommunaute.edito.models import Edito
from lacommunaute.users.factories import UserFactory


faker = Faker()


class EditoFactory(factory.django.DjangoModelFactory):
    poster = factory.SubFactory(UserFactory)
    title = faker.sentence()
    content = faker.text()
    url = faker.url()
    link_title = faker.sentence()

    class Meta:
        model = Edito

    class Params:
        for_snapshot = factory.Trait(
            title="Snapshot title",
            content="Snapshot *content*",
            url="https://example.com",
            link_title="Snapshot link title",
        )
