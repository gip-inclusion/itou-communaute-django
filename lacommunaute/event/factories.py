import datetime

import factory
from factory import fuzzy
from faker import Faker

from lacommunaute.event.models import Event
from lacommunaute.users.factories import UserFactory


faker = Faker()


class EventFactory(factory.django.DjangoModelFactory):
    poster = factory.SubFactory(UserFactory)
    name = factory.Faker("name")
    date = fuzzy.FuzzyDate(datetime.date(2023, 5, 17), datetime.date(2023, 7, 16))
    time = datetime.time(10)
    end_date = factory.LazyAttribute(lambda o: o.date)
    end_time = factory.LazyAttribute(lambda o: o.time.replace(hour=(o.time.hour + 1) % 24))
    location = faker.url()
    description = factory.LazyAttribute(lambda t: faker.text(max_nb_chars=200))

    class Meta:
        model = Event
