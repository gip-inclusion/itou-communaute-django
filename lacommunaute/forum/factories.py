import factory
from machina.core.db.models import get_model
from machina.test.factories.forum import ForumFactory as BaseForumFactory

from lacommunaute.forum.models import Forum
from lacommunaute.users.factories import GroupFactory


ForumPermission = get_model("forum_permission", "ForumPermission")
UserForumPermission = get_model("forum_permission", "UserForumPermission")


class ForumFactory(BaseForumFactory):
    type = Forum.FORUM_POST
    members_group = factory.SubFactory(GroupFactory, name=factory.SelfAttribute("..name"))
    name = factory.Faker("name")
    description = factory.Faker("sentence", nb_words=10)

    @factory.post_generation
    def with_public_perms(self, create, extracted, **kwargs):
        if not create or not extracted:
            return

        perms = [
            "can_see_forum",
            "can_read_forum",
            "can_start_new_topics",
            "can_reply_to_topics",
            "can_edit_own_posts",
            "can_delete_own_posts",
            "can_post_without_approval",
        ]

        anonymous_authorized_perms = [
            UserForumPermission(
                anonymous_user=True, authenticated_user=False, permission=permission, has_perm=True, forum=self
            )
            for permission in ForumPermission.objects.filter(codename__in=perms)
        ]

        authentified_authorized_perms = [
            UserForumPermission(
                anonymous_user=False, authenticated_user=True, permission=permission, has_perm=True, forum=self
            )
            for permission in ForumPermission.objects.filter(codename__in=perms)
        ]
        UserForumPermission.objects.bulk_create(anonymous_authorized_perms + authentified_authorized_perms)
