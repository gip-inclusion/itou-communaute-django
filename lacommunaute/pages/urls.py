from django.contrib.sitemaps.views import sitemap
from django.urls import path
from django.views.generic.base import TemplateView

from lacommunaute.pages import sitemaps, views


sitemaps = {
    "pages": sitemaps.PagesSitemap,
    "forum": sitemaps.ForumSitemap,
    "topic": sitemaps.TopicSitemap,
}

app_name = "pages"

urlpatterns = [
    path("", views.HomeView.as_view(), name="home"),
    path("contact/", views.contact, name="contact"),
    path("statistiques/", views.StatistiquesPageView.as_view(), name="statistiques"),
    path("accessibilite/", views.accessibilite, name="accessibilite"),
    path("mentions-legales/", views.mentions_legales, name="mentions_legales"),
    path("landing-pages/", views.LandingPagesListView.as_view(), name="landing_pages"),
    path("sentry-debug/", views.trigger_error, name="sentry_debug"),
    path("robots.txt", TemplateView.as_view(template_name="robots.txt", content_type="text/plain"), name="robots"),
    path("sitemap.xml", sitemap, {"sitemaps": sitemaps}, name="django.contrib.sitemaps.views.sitemap"),
]
