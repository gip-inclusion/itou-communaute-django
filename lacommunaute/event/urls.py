from django.urls import path

from lacommunaute.event.views import (
    EventCreateView,
    EventDeleteView,
    EventDetailView,
    EventListView,
    EventMonthArchiveView,
    EventUpdateView,
)


app_name = "event"


urlpatterns = [
    path("create/", EventCreateView.as_view(), name="create"),
    path("<int:pk>/", EventDetailView.as_view(), name="detail"),
    path("<int:pk>/update/", EventUpdateView.as_view(), name="update"),
    path("<int:pk>/delete/", EventDeleteView.as_view(), name="delete"),
    path("myevents/", EventListView.as_view(), name="myevents"),
    path("events/<int:year>/<int:month>/", EventMonthArchiveView.as_view(), name="month"),
    path("events/", EventMonthArchiveView.as_view(), name="current"),
]
