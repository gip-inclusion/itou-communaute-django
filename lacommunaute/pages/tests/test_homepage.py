import pytest  # noqa
from django.urls import reverse

from lacommunaute.forum.enums import Kind as ForumKind
from lacommunaute.forum.factories import ForumFactory
from lacommunaute.forum_conversation.factories import PostFactory, TopicFactory


def test_context_data(client, db):
    topic = TopicFactory(with_post=True, forum=ForumFactory())
    news = TopicFactory(with_post=True, forum=ForumFactory(kind=ForumKind.NEWS))
    article = ForumFactory(parent=ForumFactory(type=1))

    disapproved_topic = TopicFactory(with_post=True, forum=ForumFactory())
    disapproved_topic.approved = False
    disapproved_topic.save()
    TopicFactory(with_post=True, forum=ForumFactory(kind=ForumKind.PRIVATE_FORUM))

    url = reverse("pages:home")

    response = client.get(url)
    assert response.status_code == 200

    assert response.context_data["topics_public"].get() == topic
    assert response.context_data["topics_newsfeed"].get() == news
    assert response.context_data["forums_category"].get() == article


def test_new_topics_order(client, db):
    topic1 = TopicFactory(with_post=True, forum=ForumFactory())
    topic2 = TopicFactory(with_post=True, forum=ForumFactory())
    url = reverse("pages:home")

    response = client.get(url)
    assert response.status_code == 200
    assert list(response.context_data["topics_public"]) == [topic2, topic1]

    PostFactory(topic=topic1)

    response = client.get(url)
    assert response.status_code == 200
    assert list(response.context_data["topics_public"]) == [topic2, topic1]
