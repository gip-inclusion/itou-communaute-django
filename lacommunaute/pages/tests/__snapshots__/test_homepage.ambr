# serializer version: 1
# name: test_page_title_header_footer[homepage_footer]
  '''
  <footer id="footer" role="contentinfo">
      <section class="s-footer-quick">
          <div class="s-footer-quick__container container">
              <div class="s-footer-quick__row row">
                  <div class="s-footer-quick__col col-12 col-md-6 col-lg-auto">
                      <strong>Accès rapides</strong>
                      <div class="s-footer-quick__gridlist">
                          <ul>
                              <li>
                                  <a href="/topics/">Espace d'échanges</a>
                              </li>
                              <li>
                                  <a href="/documentation/">Documentation</a>
                              </li>
                              <li>
                                  <a href="/calendar/events/">Évènements</a>
                              </li>
                              <li>
                                  <a href="/search/">Chercher dans la communauté</a>
                              </li>
                          </ul>
                          <ul>
                              <li>
                                  <a href="/partenaires/">Nos partenaires</a>
                              </li>
                              <li>
                                  <a href="/surveys/dsp/create/">Diagnostic Parcours IAE</a>
                              </li>
                              <li>
                                  <a href="/members/seekers/">Mon stage CIP</a>
                              </li>
                          </ul>
                      </div>
                  </div>
                  <div class="s-footer-quick__col col-12 col-md-6 col-lg-auto">
                      <strong>Liens externes</strong>
                      <ul>
                          <li>
                              <a class="matomo-event" data-matomo-action="clic" data-matomo-category="promotion-partenaires" data-matomo-option="inclusion-demain" href="https://www.inclusion-demain.fr/" target="_blank">inclusion-demain.fr</a>
                          </li>
                          <li>
                              <a aria-label="glossaire des mots de l'inclusion" class="matomo-event" data-matomo-action="glossaire" data-matomo-category="landing" data-matomo-option="footer" href="/glossaire/" target="_blank">Glossaire de l'inclusion</a>
                          </li>
                          <li>
                              <a aria-label="Posez votre question sur la page du support des emplois de l'inclusion (lien externe)" class="matomo-event" data-matomo-action="site_emplois" data-matomo-category="support" data-matomo-option="footer" href="https://documentation.inclusion.beta.gouv.fr/aide/emplois/" target="_blank">Support des emplois de l'inclusion</a>
                          </li>
                          <li>
                              <a aria-label="Github (lien externe)" href="https://github.com/betagouv/itou-communaute-django" target="_blank">Github</a>
                          </li>
                      </ul>
                  </div>
                  <div class="s-footer-quick__col col-12 col-lg-auto">
                      <strong>Réseaux sociaux</strong>
                      <ul>
                          <li>
                              <a aria-label="La plateforme de l'inclusion sur sur Linkedin (lien externe)" href="https://www.linkedin.com/company/inclusion-gouv/" target="_blank"><i aria-hidden="true" class="ri-linkedin-box-fill text-dark"></i> LinkedIn</a>
                          </li>
                          <li>
                              <a aria-label="La communaute de l'inclusion sur Facebook (lien externe)" href="https://www.facebook.com/communaute.inclusion.gouv/" target="_blank"><i aria-hidden="true" class="ri-facebook-box-fill text-dark"></i> Facebook</a>
                          </li>
                          <li>
                              <a aria-label="La plateforme de l'inclusion sur Youtube (lien externe)" href="https://www.youtube.com/@LaPlateformedelinclusion/videos" target="_blank"><i aria-hidden="true" class="ri-youtube-fill text-dark"></i> Youtube</a>
                          </li>
                      </ul>
                  </div>
              </div>
          </div>
      </section>
      <section class="s-footer-gip">
          <div class="s-footer-gip__container container">
              <div class="s-footer-gip__row row">
                  <div class="s-footer-gip__col s-footer-gip__col--logo col-12 col-md-6 col-lg-auto">
                      <a aria-label="Découvrez les services de la plateforme de l'inclusion (lien externe)" href="https://inclusion.beta.gouv.fr/" target="_blank">
                          <img alt="Logo de la Plateforme de l'inclusion" height="80" src="/static/vendor/theme-inclusion/images/logo-plateforme-inclusion.svg"/>
                      </a>
                  </div>
                  <div class="s-footer-gip__col col-12 col-md-6 col-lg">
                      <p class="mb-1">
                          <strong>Ce service fait partie de la Plateforme de l’inclusion</strong>
                      </p>
                      <p class="m-0">
                          Découvrez les outils qui portent l'inclusion au
                          <br class="d-none d-md-inline"/>
                          coeur de leur service. A chaque service, son objectif.
                          <br/>
                          <a aria-label="Découvrez les services de la plateforme de l'inclusion (lien externe)" class="has-external-link" href="https://inclusion.beta.gouv.fr/" target="_blank">Découvrez nos services</a>
                      </p>
                  </div>
                  <div class="s-footer-gip__col s-footer-gip__col--links col-12 col-lg-auto">
                      <ul>
                          <li>
                              <a aria-label="Incubateur de services publics numériques — beta.gouv.fr (lien externe)" href="https://beta.gouv.fr/" target="_blank">beta.gouv.fr</a>
                          </li>
                          <li>
                              <a aria-label="Data.gouv.fr (lien externe)" href="https://www.data.gouv.fr/fr/" target="_blank">data.gouv.fr</a>
                          </li>
                          <li>
                              <a aria-label="Simplifiez le partage et la circulation des données administratives grace à api.gouv.fr (lien externe)" href="https://api.gouv.fr/" target="_blank">api.gouv.fr</a>
                          </li>
                      </ul>
                  </div>
              </div>
          </div>
      </section>
      <section class="s-footer-legal">
          <div class="s-footer-legal__container container">
              <div class="s-footer-legal__row row">
                  <div class="s-footer-legal__col col-12">
                      <ul>
                          <li>
                              <a aria-label="Mentions légales" href="/mentions-legales/">Mentions légales</a>
                          </li>
                          <li>
                              <a aria-label="Accessibilité" href="/accessibilite/">Accessibilité : non conforme</a>
                          </li>
                          <li>
                              <a aria-label="Politique de confidentialité" href="/politique-de-confidentialite/">Politique de confidentialité</a>
                          </li>
                          <li>
                              <a href="/statistiques/">Statistiques</a>
                          </li>
                      </ul>
                      <p>
                          Sauf mention contraire, tous les contenus de ce site sont sous licence <a aria-label="Licence ouverte Etalab (lien externe)" href="https://github.com/etalab/licence-ouverte/blob/master/LO.md" target="_blank">etalab-2.0</a>
                      </p>
                  </div>
              </div>
          </div>
      </section>
  </footer>
  '''
# ---
# name: test_page_title_header_footer[homepage_header]
  '''
  <header id="header" role="banner">
      <section class="s-header">
          <div class="s-header__container container">
              <div class="s-header__row row">
                  <div class="s-header__col s-header__col--logo-ministere col col-md-auto d-flex align-items-center pe-0">
                      <a class="s-header__logo-ministere" href="/">
                          <img alt="Rébublique Française" src="/static/vendor/theme-inclusion/images/logo-republique-francaise.svg"/>
                      </a>
                  </div>
                  <div class="s-header__col s-header__col--logo-service col-auto d-flex align-items-center px-0">
                      <a href="/">
                          <img alt="La communauté de l'inclusion" height="90" src="/static/vendor/theme-inclusion/images/logo-communaute-inclusion.svg"/>
                      </a>
                  </div>
                  <div class="s-header__col s-header__col--nav col d-none d-lg-inline-flex d-lg-flex align-items-center justify-content-end pe-0 pe-xl-3">
                      <nav aria-label="Navigation principale" id="nav-primary" role="navigation">
                          <ul>
                              <li id="header_search_form">
                                  <form action="/search/" name="header_search_form">
                                      <div class="input-group input-group-rounded-pill">
                                          <input aria-label="Chercher dans la communauté" class="form-control w-300px" name="q" placeholder="Chercher dans la communauté" type="search"/>
                                          <div class="input-group-text p-0">
                                              <button class="btn btn-link btn-ico matomo-event" data-matomo-action="search" data-matomo-category="engagement" data-matomo-option="submit_query_header" type="submit">
                                                  <i aria-label="Chercher dans la communauté" class="ri-search-line font-weight-bold"></i>
                                              </button>
                                          </div>
                                      </div>
                                  </form>
                              </li>
                              
                                  <li>
                                      <a class="btn btn-inclusion-connect" href="/inclusion_connect/authorize" rel="nofollow">
                                          <img alt="Se connecter | S'inscrire" height="37" src="/static/vendor/theme-inclusion/images/logo-inclusion-connect-two-lines.svg"/>
                                      </a>
                                  </li>
                              
                          </ul>
                      </nav>
                  </div>
                  <div class="s-header__col s-header__col--burger col col-lg-auto d-flex align-items-center justify-content-end d-xl-none ps-0">
                      <button aria-controls="offcanvasNav" aria-label="Ouvrir le menu de navigation principale pour mobile" class="btn btn-link btn-ico pe-0" data-bs-target="#offcanvasNav" data-bs-toggle="offcanvas" type="button">
                          <i aria-hidden="true" class="ri-menu-line" id="open"></i>
                          <span>Menu</span>
                      </button>
                  </div>
              </div>
          </div>
      </section>
      <section class="s-postheader d-none d-xl-block">
          <div class="s-postheader__container container">
              <div class="s-postheader__row row">
                  <div class="s-postheader__col col-12 d-flex align-items-center">
                      <nav aria-label="Navigation secondaire" role="navigation">
                          <ul>
                              <li>
                                  <a class="is-active" href="/">Accueil</a>
                              </li>
                              <li>
                                  <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="topics" href="/topics/">Espace d'échanges</a>
                              </li>
                              <li>
                                  <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="documentation_header" href="/documentation/">Documentation</a>
                              </li>
                              <li>
                                  <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="events" href="/calendar/events/">Évènements</a>
                              </li>
                              <li>
                                  <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="dsp" href="/surveys/dsp/create/">
                                      Diagnostic Parcours IAE
                                  </a>
                              </li>
                          </ul>
                      </nav>
                  </div>
              </div>
          </div>
      </section>
      <div aria-label="Navigation principale pour mobile" class="offcanvas" id="offcanvasNav">
          <div class="offcanvas-header">
              <h4 class="h5 mb-0 btn-ico align-items-center" id="offcanvasApplyFiltersLabel">
                  <i aria-hidden="true" class="ri-menu-line"></i>
                  <span>Menu</span>
              </h4>
              <button aria-label="Fermer" class="btn-link px-2 me-n2" data-bs-dismiss="offcanvas" type="button">
                  <i aria-hidden="true" class="ri-close-line"></i>
              </button>
          </div>
          <div class="offcanvas-body">
              <nav aria-label="Navigation secondaire pour mobile" role="navigation">
                  <form action="/search/" class="mb-4" name="header_search_form">
                      <div class="input-group input-group-rounded-pill">
                          <input aria-label="Chercher dans la communauté" class="form-control" name="q" placeholder="Chercher dans la communauté" type="search"/>
                          <div class="input-group-text p-0">
                              <button class="btn btn-link btn-ico matomo-event" data-matomo-action="search" data-matomo-category="engagement" data-matomo-option="submit_query_header" type="submit">
                                  <i aria-label="Chercher dans la communauté" class="ri-search-line font-weight-bold"></i>
                              </button>
                          </div>
                      </div>
                  </form>
                  <ul class="nav nav-tabs flex-column">
                      <li class="nav-item">
                          <a class="nav-link active" href="/">Accueil</a>
                      </li>
                      <li class="nav-item">
                          <a class="nav-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="topics" href="/topics/">Espace d'échanges</a>
                      </li>
                      <li class="nav-item">
                          <a class="nav-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="documentation_header" href="/documentation/">Documentation</a>
                      </li>
                      <li>
                          <a class="nav-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="events" href="/calendar/events/">Évènements</a>
                      </li>
                      <li>
                          <a class="nav-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="dsp" href="/surveys/dsp/create/">
                              Diagnostic Parcours IAE
                          </a>
                      </li>
                  </ul>
              </nav>
          </div>
          <div class="offcanvas-footer flex-column align-items-stretch">
              <nav aria-label="Menu de navigation principale pour mobile" role="navigation">
                  
                      <div>
                          <a class="btn btn-inclusion-connect" href="/inclusion_connect/authorize" rel="nofollow">
                              <img alt="Se connecter | S'inscrire" class="img-fluid" src="/static/vendor/theme-inclusion/images/logo-inclusion-connect-one-line.svg"/>
                          </a>
                      </div>
                  
              </nav>
          </div>
      </div>
  </header>
  '''
# ---
# name: test_page_title_header_footer[homepage_title]
  '''
  <title>
                  Accueil - La communauté de l'inclusion
              </title>
  '''
# ---
